### DISCUSSION TITLE
Objectify Find

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2741] [Suggestions]
Have a `Find` object that works like effects, where you set its fields before triggering an operation. This would allow
- Instanced find operations with their own whitelists and results
- Restructuring these new find operations to be more idiomatic
- Deprecating the somewhat confusing current way of finding things

**Comment by vurvdev** [comment]
The worst thing about the find library is the quota. A new system having dynamic op counts based on the amount of filters would be great.

Ideally as I sort of joked on discord, you'd just be able to do the filtering yourself with lambdas, but we don't have them yet.

```lua
local PropsInSphere = allEntities()
    :filter(E => E:pos():distance(Point) < 500)
    :filter(E:type() == "prop_physics")
```

So I guess if this is done without lambdas, this would be my ideal:
```lua
local PropsInSphere = findAllEntities() # Returns a `findquery` type
    :inListOfEntities(Array)
    :inSphere(Pos, Radius)
    :notInSphere(Pos, Radius)
    :withClass(String)
    :withoutClass(String)
    :withOwner(Entity)
    :withoutOwner(Entity)
    :query() # returns array of results. should store the query object to run this repeatedly.
 ```
 
 Basically for the naming scheme:
 * `in*` methods for the current findIn*
 * `with*` methods for including entities with specific property, like class, owner, etc
 * `without*` methods for negated version of `with*` methods
 * `not*` methods for negated versions of the created methods

**Reply by Denneisk** [reply]
That's pretty solid.

**Reply by vurvdev** [reply]
Turns out E2 bans use of this syntax since methodcalls have a restriction on whitespace.

Will need to change that

**Reply by Denneisk** [reply]
Who needs it? Just have one really long line instead :^)

**Comment by Fasteroid** [comment]
SEX

**Comment by Fasteroid** [comment]
YES PLEASE
